Mathlib/Combinatorics/SimpleGraph/Clique.lean

```diff
@@ -330,32 +509,55 @@ def cliqueFinset (n : ℕ) : Finset (Finset α) :=
   univ.filter <| G.IsNClique n
 #align simple_graph.clique_finset SimpleGraph.cliqueFinset
 
+@[simp, norm_cast]
+theorem coe_cliqueFinset (n : ℕ) : (G.cliqueFinset n : Set (Finset α)) = G.cliqueSet n := by
+  ext; simp [cliqueFinset]
+#align simple_graph.coe_clique_finset SimpleGraph.coe_cliqueFinset
+
+variable {G}
+
+@[simp]
 theorem mem_cliqueFinset_iff : s ∈ G.cliqueFinset n ↔ G.IsNClique n s :=
   mem_filter.trans <| and_iff_right <| mem_univ _
 #align simple_graph.mem_clique_finset_iff SimpleGraph.mem_cliqueFinset_iff
 
-@[simp]
-theorem coe_cliqueFinset (n : ℕ) : (G.cliqueFinset n : Set (Finset α)) = G.cliqueSet n :=
-  Set.ext fun _ ↦ mem_cliqueFinset_iff _
-#align simple_graph.coe_clique_finset SimpleGraph.coe_cliqueFinset
-
```

Comment:
This looks like a nicer proof than the new one, and I don't see it in the dashboard diff; was this accidental?

Reply:
Accidental yes. This file was already ported.

Reply:
This is because `mem_cliqueFinset_iff` has now moved below to take advantage of the `variable {G}` I added.

Reply:
I think you should optimize for the proof itself, not minimizing `variable` noise. You might want `variable {G} in`, (or `(G) in`), now that we're in Lean 4.